islandora.module

```diff
@@ -331,6 +333,191 @@ function islandora_form_alter(&$form, FormStateInterface $form_state, $form_id)
       }
     }
   }
+
+  $form_object = $form_state->getFormObject();
+  $utils = \Drupal::service('islandora.utils');
+  $config = \Drupal::config('islandora.settings')->get('delete_media_and_files');
+
+  if ($config == 1 && $form_object instanceof EntityForm) {
+    $entity = $form_object->getEntity();
+
+    if ($entity->getEntityTypeId() == "node" && $utils->isIslandoraType($entity->getEntityTypeId(), $entity->bundle()) && strpos($form['#form_id'], 'delete_form') !== FALSE) {
+      $medias = $utils->getMedia($form_state->getFormObject()->getEntity());
+      if (count($medias) != 0) {
+        $form['delete_associated_content'] = [
+          '#type' => 'checkbox',
+          '#title' => t('Delete all associated medias and nodes'),
+        ];
+
+        $media_list = [];
+
+        foreach ($medias as $media) {
+          $media_list[] = $media->getName();
+        }
+
+        $form['media_items'] = [
```

Comment:
If you want (don't have to for this PR) you could add `#states` to this so the `item_list` only appears when the user toggles the `Delete all` functionality.

Given you are using `#theme` you'd need to wrap it in a container so something like the below _should_ work:

```
$form['container'] = [
  '#type' => 'container',
  '#states' => [
    'visible' => [
      ':input[name="delete_associated_content"]' => ['checked' => TRUE],
    ],
  ],
];
$form['container']['media_items'] = [
  '#theme' => 'item_list',
  '#type' => 'ul',
  '#items' => $media_list,
  '#attributes' => ['class' => ['islandora-media-items']],
  '#wrapper_attributes' => ['class' => ['container']],
  '#attached' => [
    'library' => [
      'islandora/drupal.islandora.theme_css',
    ],
  ],
];
```